appservice/src/SiteWrapper.ts

```diff
@@ -195,6 +200,56 @@ export class SiteWrapper {
         return await this.updateScmType(client, config, newScmType);
     }
 
+    /**
+     * Starts the log-streaming service. Call 'dispose()' on the returned object when you want to stop the service.
+     */
+    public async startStreamingLogs(client: KuduClient, actionHandler: AzureActionHandler, outputChannel: vscode.OutputChannel, path: string = ''): Promise<vscode.Disposable | undefined> {
+        const httpRequest: WebResource = new WebResource();
+        await new Promise((resolve: () => void, reject: (err: Error) => void): void => {
+            client.credentials.signRequest(httpRequest, (err: Error) => {
+                if (err) {
+                    reject(err);
+                } else {
+                    resolve();
+                }
+            });
+        });
+
+        const requestApi: request.RequestAPI<request.Request, request.CoreOptions, {}> = request.defaults(httpRequest);
+        let disposable: vscode.Disposable | undefined;
+        // Intentionally setting up a separate telemetry event and not awaiting the result here since log stream is a long-running action
+        // tslint:disable-next-line:no-floating-promises
+        actionHandler.callWithTelemetry('appService.streamingLogs', async () => {
```

Comment:
Do we want all of our telemetry logs to come in with the label appsService.streamingLogs?  Can we make it context based?

Reply:
I mean both extensions are for app services, so the label is accurate. I think it's fine to be the same name. After all, it's the same code and we still have unique 'startStreamingLogs' events for each extension